docs(architecture): SCRUM-45 Document a modular monolith (#11)
CPAprince/Twitter: docs/modular-monolith.md

```text 
src/

> ├── Iam/

Reviewer: ```suggestion
├── IAM/
```

│   ├── resources/
│   │   └── config/
│   │       ├── doctrine/
│   │       │   └── *.orm.xml
│   │       ├── services.yaml
│   │       ├── security.yaml
│   │       └── routes.yaml
│   ├── UI/

> │   │   └── Rest/

Reviewer: ```suggestion
│   │   └── REST/
```

> │   │       ├── Request/

Reviewer: ```suggestion
```

│   │       ├── Controller/
│   │       └── Exception/
│   ├── Application/

> │   │   ├── Validator/  # (interfaces) e.g., UniqueEmailCheckerInterface (define contract here) 

Reviewer: ```suggestion
```

> │   │   ├── Handler/    # Validate DTO, call validators/interfaces, orchestrate use case

Reviewer: ```suggestion
│   │   ├── UseCase/    # Stores handlers, queries, comamnds, DTO
```

> │   │   ├── Command/

Reviewer: ```suggestion
```

> │   │   ├── Dto/

Reviewer: ```suggestion
```

> │   │   ├── Exception/

Reviewer: ```suggestion
```

> │   │   └── Service/

Reviewer: ```suggestion
```

> │   ├── Domain/

Reviewer: @adanyl0v Domain -> Name -> Model -> (Entity, Aggregate, Value object, RepositoryInterafce). Examples:
- Domain -> User -> Model -> User.php
- Domain -> User -> Model -> Email.php
- Domain -> User -> Model -> UserRepository.php

│   │   ├── Model/

> │   │   ├── ValueObject/

Reviewer: ```suggestion
```

> │   │   ├── Repository/

Reviewer: ```suggestion
```

> │   │   └── Exception/

Reviewer: ```suggestion
```

│   └── Infrastructure/
│       ├── Exception/
│       ├── EventSubscriber/
│       │   ├── Doctrine/
│       │   │   └── HashPasswordSubscriber.php
│       │   └── Symfony/
│       │       └── ExceptionToHttpResponseSubscriber.php
│       ├── Persistence/
│       │   └── MySQL
│       │       ├── Repository/
│       │       └── Migrations/
│       └── Auth/
│           ├── Authenticator/      # Guard/ is soft-deprecated since Symfony 5.3
│           │   └── JsonLoginAuthenticator.php
│           ├── UserProvider.php    # implements Symfony user provider
│           └── UserAdapter.php     # maps Domain\Model\User -> Symfony user
├── Profile/
│   └── ...
└── Tweet/

>     └── ...

Reviewer: ```suggestion
    └── ...
└── Shared/
    └── ...
```

```

## Modular monolith structure

> The structure is solid and modular: each **feature module** (Iam, Profile, Tweet) follows the same vertical-slice

Reviewer: ```suggestion
The structure is solid and modular: each **module** (Iam, Profile, Tweet) follows the same vertical-slice
```

> pattern (Resources / UI / Application / Domain / Infrastructure). That’s exactly what you want for a modular monolith:

Reviewer: ```suggestion
pattern (UI / Application / Domain / Infrastructure). That’s exactly what you want for a modular monolith:
```

clear boundaries, easy testing and straightforward extraction later. Below — precise, directory-by-directory
explanation (what it does, why it’s here, and short pitfalls/guidance).

### `src/` (top level)

> **What:** root for all application code, organized by feature/module.

Reviewer: ```suggestion
**What:** root for all application code, organized by modules.
```

> **Why:** encourages feature ownership, reduces cross-cutting coupling, simplifies CI/test scoping.

Reviewer: ```suggestion
**Why:** encourages context ownership, reduces cross-cutting coupling, simplifies CI/test scoping.
```

**Pitfalls / guidance:** keep a small `Shared` area for truly cross-cutting contracts (not business logic). Add

> top-level `tests/` and `migrations/` or document migration ownership per module.

Reviewer: ```suggestion
```

> ### `src/Iam/resources/config/`

Reviewer: @adanyl0v Not the time for that. Remove it for now.
